helix-core/src/main/java/org/apache/helix/cloud/event/helix/DefaultCloudEventCallbackImpl.java

```diff
@@ -19,22 +19,27 @@
  * under the License.
  */
 
+import java.util.List;
+
 import org.apache.helix.HelixManager;
-import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+import org.apache.helix.constants.InstanceConstants;
+import org.apache.helix.util.InstanceValidationUtil;
 
 /**
  * A default callback implementation class to be used in {@link HelixCloudEventListener}
  */
 public class DefaultCloudEventCallbackImpl {
+  public static final String REASON = "CloudEvent";
```

Comment:
Let's have a detailed reason and put to maintenance mode / instance disable reason?

Reply:
Reason added. Please review.